File: src/main/java/com/polygloat/dtos/response/PermissionDTO.java
package com.polygloat.dtos.response;

import com.polygloat.model.Permission;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PermissionDTO {
    private Long id;

    private Permission.RepositoryPermissionType type;

    private String username;

    private Long userId;

    private String userFullName;

    public static PermissionDTO fromEntity(Permission entity) {
        return new PermissionDTO(entity.getId(), entity.getType(), entity.getUser().getUsername(), entity.getUser().getId(), entity.getUser().getName());
    }
}

File: src/main/kotlin/io/tolgee/dtos/cacheable/PermissionDto.kt
package io.tolgee.dtos.cacheable

import io.tolgee.model.Permission
import java.io.Serializable

data class PermissionDto(
  val id: Long,
  val userId: Long?,
  val invitationId: Long?,
  val type: Permission.ProjectPermissionType,
  val projectId: Long
) : Serializable {
  companion object {
    fun fromEntity(entity: Permission) = PermissionDto(
      userId = entity.user?.id,
      invitationId = entity.invitation?.id,
      id = entity.id,
      type = entity.type,
      projectId = entity.project.id
    )
  }
}
